File: lab3/app/src/main/java/com/example/pam1/MainActivity.java
package com.example.pam1;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import java.util.regex.Pattern;

public class MainActivity extends AppCompatActivity {

    private EditText edt0;
    private EditText edt1;
    private EditText edt2;
    private Button butt;
    private TextView res;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        edt0 = findViewById(R.id.licz1);
        edt1 = findViewById(R.id.znak);
        edt2 = findViewById(R.id.licz2);
        butt = findViewById(R.id.add);
        ;
        res = findViewById(R.id.liczres);

        edt0.setOnFocusChangeListener(myOwnOnFocusChangeListener);
        edt1.setOnFocusChangeListener(myOwnOnFocusChangeListener);
        edt2.setOnFocusChangeListener(myOwnOnFocusChangeListener);
        butt.setOnClickListener(myOwnClick);
        butt.setOnLongClickListener(loggg);
    }

    private View.OnClickListener myOwnClick = v -> {
        switch (edt1.getText().toString()) {
            case "+":
                add();
                break;
            case "-":
                sub();
                break;
            case "*":
                mult();
                break;
            case "/":
                div();
                break;
            default:
                res.setText("WTF");
                break;
        }
    };

    private View.OnLongClickListener loggg = v -> {
        showToast();
        return false;
    };

    private void showToast() {
        Toast.makeText(MainActivity.this,"Programowanie Aplikacji Mobilnych",
                Toast.LENGTH_SHORT).show();
    }

    private View.OnFocusChangeListener myOwnOnFocusChangeListener = (v, hasFocus) -> checkInput();

    public void checkInput() {
        if (!edt1.getText().toString().contains("+") && !edt1.getText().toString().contains("-") && !edt1.getText().toString().contains("*") && !edt1.getText().toString().contains("/")) {
            res.setText("math sign should be one of: +, -, *, /");
            butt.setVisibility(Button.INVISIBLE);
        } else {
            if (edt0.getText().length() > 0 && edt2.getText().length() > 0){
                if (!edt0.getText().toString().contains(" ") && !edt2.getText().toString().contains(" ") && !Pattern.matches("[a-zA-Z]+", edt0.getText()) && !Pattern.matches("[a-zA-Z]+", edt2.getText()) && !Pattern.matches("[а-яА-Я]+", edt0.getText()) && !Pattern.matches("[а-яА-Я]+", edt2.getText()))
                    butt.setVisibility(Button.VISIBLE);
                else
                    butt.setVisibility(Button.INVISIBLE);
            }
            else
                butt.setVisibility(Button.INVISIBLE);
        }

    }

    public void add() {
        EditText l1 = findViewById(R.id.licz1);
        EditText l2 = findViewById(R.id.licz2);
        TextView result = findViewById(R.id.liczres);

        result.setText(l1.getText() + " + " + l2.getText() + " = " + String.valueOf(Float.parseFloat(l1.getText().toString()) + Float.parseFloat(l2.getText().toString())));
    }

    public void sub() {
        EditText l1 = findViewById(R.id.licz1);
        EditText l2 = findViewById(R.id.licz2);
        TextView result = findViewById(R.id.liczres);

        result.setText(l1.getText() + " - " + l2.getText() + " = " + String.valueOf(Float.parseFloat(l1.getText().toString()) - Float.parseFloat(l2.getText().toString())));
    }

    public void mult() {
        EditText l1 = findViewById(R.id.licz1);
        EditText l2 = findViewById(R.id.licz2);
        TextView result = findViewById(R.id.liczres);

        result.setText(l1.getText() + " * " + l2.getText() + " = " + String.valueOf(Float.parseFloat(l1.getText().toString()) * Float.parseFloat(l2.getText().toString())));
    }

    public void div() {
        EditText l1 = findViewById(R.id.licz1);
        EditText l2 = findViewById(R.id.licz2);
        TextView result = findViewById(R.id.liczres);

        if (Integer.parseInt(l2.getText().toString()) == 0) {
            result.setText("Cant div on 0");
            return;
        }

        result.setText(l1.getText() + " / " + l2.getText() + " = " + String.valueOf(Float.parseFloat(l1.getText().toString()) / Float.parseFloat(l2.getText().toString())));
    }
}

File: lab5/app/src/main/java/com/example/pam1/MainActivity.kt
package com.example.pam1

import android.os.Bundle
import android.view.View
import android.view.View.OnFocusChangeListener
import android.view.View.OnLongClickListener
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import java.util.regex.Pattern

class MainActivity : AppCompatActivity(), View.OnClickListener, View.OnFocusChangeListener {
    private lateinit var edt0: EditText
    private lateinit var edt1: EditText
    private lateinit var edt2: EditText
    private lateinit var lic1: EditText
    private lateinit var lic2: EditText
    private lateinit var butt: Button
    private lateinit var buttSort: Button
    private lateinit var res: TextView
    private lateinit var reska: TextView
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        edt0 = findViewById(R.id.licz1)
        edt1 = findViewById(R.id.znak)
        edt2 = findViewById(R.id.licz2)
        lic1 = findViewById(R.id.l0)
        lic2 = findViewById(R.id.l1)
        butt = findViewById(R.id.add)
        buttSort = findViewById(R.id.sort)
        res = findViewById(R.id.liczres)
        reska = findViewById(R.id.reska)

        edt0.onFocusChangeListener = this
        edt1.onFocusChangeListener = this
        edt2.onFocusChangeListener = this

        butt.setOnClickListener(this)
        buttSort.setOnClickListener(this)
    }

    override fun onFocusChange(p0: View?, p1: Boolean) { checkInput() }

    override fun onClick(v: View) {
        when(v.id)
        {
            R.id.add -> res.text = calc().toString()
            R.id.sort -> reska.text = "Min: " + sort().minOrNull().toString() + " Max: " + sort().maxOrNull().toString() + " AVG: " + sort().average().toString()
        }
    }

    private fun sort(): List<Float> {
        return lic2.text.split(", ").map { it.trim().toFloat() }
    }

    private fun calc(): Float{
        return when (edt1.text.toString()) {
            "+" -> add()
            "-" -> sub()
            "*" -> mult()
            "/" -> div()
            else -> 404F
        }
    }

    private fun checkInput() {
        if (!edt1.text.toString().contains("+") && !edt1.text.toString().contains("-") && !edt1.text.toString().contains("*") && !edt1.text.toString().contains("/")) {
            "math sign should be one of: +, -, *, /".also { res.text = it }
            butt.visibility = Button.INVISIBLE
        } else {
            if (edt0.text.isNotEmpty() && edt2.text.isNotEmpty()) {
                if (!edt0.text.toString().contains(" ") && !edt2.text.toString()
                        .contains(" ") && !Pattern.matches(
                        "[a-zA-Z]+",
                        edt0.text
                    ) && !Pattern.matches("[a-zA-Z]+", edt2.text) && !Pattern.matches(
                        "[а-яА-Я]+",
                        edt0.text
                    ) && !Pattern.matches("[а-яА-Я]+", edt2.text)
                ) butt.visibility = Button.VISIBLE else butt.visibility = Button.INVISIBLE
            } else butt.visibility = Button.INVISIBLE
        }
    }

    private fun add(): Float {
        val l1 = findViewById<EditText>(R.id.licz1)
        val l2 = findViewById<EditText>(R.id.licz2)
        val result = findViewById<TextView>(R.id.liczres)
        return l1.text.toString().toFloat() + l2.text.toString().toFloat()
    }

    private fun sub(): Float {
        val l1 = findViewById<EditText>(R.id.licz1)
        val l2 = findViewById<EditText>(R.id.licz2)
        return l1.text.toString().toFloat() - l2.text.toString().toFloat()
    }

    private fun mult(): Float {
        val l1 = findViewById<EditText>(R.id.licz1)
        val l2 = findViewById<EditText>(R.id.licz2)
        val result = findViewById<TextView>(R.id.liczres)
        return l1.text.toString().toFloat() * l2.text.toString().toFloat()
    }

    private fun div(): Float {
        val l1 = findViewById<EditText>(R.id.licz1)
        val l2 = findViewById<EditText>(R.id.licz2)
        val result = findViewById<TextView>(R.id.liczres)
        if (l2.text.toString().toInt() == 0) {
            "Cant div on 0".also { result.text = it }
            return 404F
        }
        return l1.text.toString().toFloat() / l2.text.toString().toFloat()
    }
}